portal/models/user.py

```diff
@@ -1487,6 +1487,19 @@ def merge_with(self, other_id):
     def promote_to_registered(self, registered_user):
         """Promote a weakly authenticated account to a registered one"""
         assert self.id != registered_user.id
+
+        # Ensure the registered user is not a power user
+        # https://jira.movember.com/browse/TN-1408
+        restricted_roles = \
+            current_app.config['RESTRICTED_FROM_PROMOTION_ROLES']
+        for restricted_role in restricted_roles:
+            if registered_user.has_role(restricted_role):
+                error_message = 'Attempted to promote temporary user {} \
+                        to registered user {} with role {}' \
+                        .format(self.id, registered_user.id, restricted_role)
+                current_app.logger.error(error_message)
+                return False
```

Comment:
I think it'd be better to raise an exception here - this is an exceptional case, and we don't want to rely on the client to check the return value (plus, they don't know *why* it failed).